### PR TITLE
[BugFix]Fix no bucket num bug when create table in shared data mode

### PR DESCRIPTION
**Pull request by abc982627271**
Fixes #issue

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java [hunk 1]
```diff
@@ -303,6 +304,9 @@ public static void checkPartitionValuesExistForAddListPartition(OlapTable olapTa
 
     public static int calBucketNumAccordingToBackends() {
         int backendNum = GlobalStateMgr.getCurrentSystemInfo().getBackendIds().size();
+        if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
+            backendNum = backendNum + GlobalStateMgr.getCurrentSystemInfo().getComputeNodeIds(false).size();
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/catalog/CatalogUtils.java:308]: add `// TODO:` or `// FIXME: `, we need to revisit this code and define a proper behavior for shared data mode.

**Comment by wyb** [comment]
better to optimize the error message if there is no be(cn) and backendNum is 0

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|1|2|50.00%|[308]|

**Review by kevincai — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 4 / 5 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|3|4|75.00%|[1600]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 3 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|2|50.00%|[1598]|
|:large_blue_circle:|com/starrocks/catalog/CatalogUtils.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|2|50.00%|[1598]|

**Review by chaoyli — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request closed without merging**